core/iwasm/libraries/libc-wasi/libc_wasi_wrapper.c

```diff
@@ -1171,6 +1203,22 @@ wasi_sock_set_recv_buf_size(wasm_exec_env_t exec_env, wasi_fd_t fd,
     return __WASI_ENOSYS;
 }
 
+static wasi_errno_t
+wasi_sock_set_recv_timeout(wasm_exec_env_t exec_env, wasi_fd_t fd,
+                           uint64_t *timeout_us)
+{
+    wasm_module_inst_t module_inst = get_module_inst(exec_env);
+    wasi_ctx_t wasi_ctx = get_wasi_ctx(module_inst);
+    struct fd_table *curfds = NULL;
+
+    if (!wasi_ctx)
+        return __WASI_EACCES;
+
+    curfds = wasi_ctx_get_curfds(module_inst, wasi_ctx);
+
+    return wasmtime_ssp_sock_set_recv_timeout(curfds, fd, *timeout_us);
```

Comment:
before passing the `timeout_us` further, we need to check do check the memory using `validate_native_addr()`. See usage of that macro in this file for examples.